core/src/Streamly/Internal/Data/Array/Mut/Type.hs

```diff
@@ -84,9 +84,10 @@ module Streamly.Internal.Data.Array.Mut.Type
 
     -- * Eliminating and Reading
 
-    -- ** To streams
-    , read
-    -- , readRev
+    -- ** Unfolds
+    , reader
+    -- , readerRev
+    , producer -- experimental
```

Comment:
There is a `toStreamD` and `toStreamK` below.
Any reason you did not rename those?